### PR TITLE
[409297][hightlighting][deprecation] Semantics different from JDTs

### PR DESCRIPTION
**Pull request by cdietrich**
[409297][hightlighting][deprecation] Semantics different from JDTs
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=409297

**Comment by cdietrich** [comment]
please review. see https://github.com/eclipse/xtext-xtend/pull/209 as well

File: org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/DeprecationUtil.java [hunk 1]
```diff
@@ -23,7 +23,8 @@
 public class DeprecationUtil {
 	public static boolean isDeprecated(JvmAnnotationTarget jvmAnnotationTarget) {
 		if (jvmAnnotationTarget instanceof JvmMember) {
-			return isDeprecatedMember((JvmMember) jvmAnnotationTarget);
+			return isDeprecatedMember((JvmMember) jvmAnnotationTarget)
```
Comment (szarnekow) [inline org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/DeprecationUtil.java]: The semantics is now the same as for #isTransitivelyDeprecated. Why doesn't the caller use that one instead?

Comment (cdietrich) [inline org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/DeprecationUtil.java]: ill have a look

**Comment by cdietrich** [comment]
i updated the pullrequest @szarnekow

**Comment by szarnekow** [comment]
I wonder if we should make it symmetric and more explicit by adding isTransitivelyDeprecated(JvmAnnotationTarget) and calling that from the highlighting calculator?

**Comment by cdietrich** [comment]
good question. can we make this somehow backwards compatible api wise?

**Comment by szarnekow** [comment]
Adding a new method is not backwards compatible. Apart from that, it would keep the behavior consistent with the prev version.

**Pull request merged**